Tella/Scenes/Settings/ViewModel/GDriveAuthViewModel.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  GDriveAuthViewModel.swift
+//  Tella
+//
+//  Created by gus valbuena on 5/22/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class GDriveAuthViewModel: ObservableObject {
+    private let gDriveRepository: GDriveRepositoryProtocol
+    private var cancellables = Set<AnyCancellable>()
+
+    init(repository: GDriveRepositoryProtocol = GDriveRepository.shared) {
+        self.gDriveRepository = repository
+    }
+
+    func handleSignIn(completion: @escaping () -> Void) {
+        Task {
+            do {
+                try await gDriveRepository.handleSignIn()
+                DispatchQueue.main.async {
+                    completion()
+                }
+            } catch let error {
+                DispatchQueue.main.async {
+                    Toast.displayToast(message: error.localizedDescription)
+                }
```

Comment:
Views should not be displayed in the viewModels: 
Maybe you can try this: 
```
    func handleSignIn() async throws {
        try await gDriveRepository.handleSignIn()
    }

```
and handle the rest in `navigateToGDriveFlow` function inside the file `ServerSelectionView` 
Ill write the proposed method in `ServerSelectionView`